### ISSUE TITLE
[Editor] Lack of DirectX 10 support crashing the editor

### ISSUE THREAD

**Issue by Eideren** [xenko3d/xenko#198]
**Release Type**: GitHub

**Version**: [Latest 3.0.0.5](https://github.com/xenko3d/xenko/commit/1cc4d2761db038ebe89f04c1787b572b9ebc6dc0)

**Platform(s)**: Editor on windows desktop

**Describe the bug**
Exception thrown crashing the application and crash reporter, caused by old graphics card, graphics emulation or deprecated drivers.

**To Reproduce**
1. Use a graphics card which doesn't support dx10+/ outdated driver / a VM with graphics emulation
2. Launch the editor.

**Expected behavior**
Not crashing / switching to a supported graphics API, if not possible present the user with a modal window describing the issue before crashing.

**Screenshots**
--

**Log and callstacks**
[Exception thrown](https://github.com/xenko3d/xenko/files/2375042/GP10%2BNotSupported.txt)

**Additional context**
This issue is a follow up of #124

**Comment by xen2** [comment]
Good point, editor needs DX10+ minimum (or even more depending on what your game is doing, i.e. tessellation).
https://github.com/xenko3d/xenko/blob/releases/3.0.0.4/sources/editor/Xenko.Editor/Engine/EmbeddedGame.cs#L22

Surprised there was no check in editor startup/load, we need something.
Two options (we might want both, (1) is probably easy to add and would solve most crashes like people trying in VM with DX9 only):
(1) Early startup check for DX10+
(2) Later check when opening an actual game with the feature level from GameSettings
